server/src/main/java/com/flipkart/varadhi/auth/DefaultAuthorizationProvider.java

```diff
@@ -127,27 +131,26 @@ private List<Pair<ResourceType, ResourceContext>> generateResourceContextHierarc
     private boolean isAuthorizedInternal(
             String subject, ResourceAction action, ResourceContext resourceContext
     ) {
-        log.debug(
-                "Checking authorization for subject [{}] and action [{}] on resource [{}]", subject, action,
-                resourceContext.policyPath()
-        );
         return getRolesForSubject(subject, resourceContext)
                 .stream().anyMatch(role -> doesActionBelongToRole(subject, role, action));
     }
 
     private Set<String> getRolesForSubject(String subject, ResourceContext resourceContext) {
-        RoleBindingNode node =
-                getAuthZService().getIAMPolicy(resourceContext.resourceType(), resourceContext.resourceId());
-        if (node == null) {
-            log.error("No roles on resource for subject {}", subject);
+        try {
+            IamPolicyRecord node =
+                    getAuthZService().getIamPolicy(resourceContext.resourceType(), resourceContext.resourceId());
+            if (node == null) {
```

Comment:
let's handle only ResourceNotFoundException instead of handling both. Also add a test to ensure getAuthZService().getIamPolicy() throws exception instead of returning null.

Reply:
removed null check
test is already added to IamPolicyServiceTest to check for the required case